app/models/post.rb

```diff
@@ -0,0 +1,16 @@
+class Post < ApplicationRecord
+    belongs_to :author, class_name: 'User'
+    has_many :comments
+    has_many :likes
+  
+    def recent_comments
+      comments.order(created_at: :desc).limit(5)
+    end
+    after_create :increment_user_posts_counter
+  
+    private
+  
+    def increment_user_posts_counter
+      author.increment!(:posts_counter)
+    end
```

Comment:
- [ ] Kindly I recommend you to use one of the [Rails Active Callbacks](https://guides.rubyonrails.org/active_record_callbacks.html) - after_save to call the #`increment_user_posts_counter`method every time a new post has been saved to database.

- `after_create` is a callback that is only triggered when a new object is created. It will not be triggered when an object is updated.

- `after_save`, on the other hand, is triggered every time an object is saved. This means it will be called both when an object is created and when it is updated.

- [x] In your case, if you use `after_create` :`increment_user_posts_counter`, the `increment_user_posts_counter` method will only be called when a new post is created. If a post is updated, the counter will not change.

- [x] If you use` after_save :increment_user_posts_counter,` the `increment_user_posts_counter `method will be called every time a post is created or updated. This might be useful if the update can change whether the post is counted or not. So kindly I recommend you to use **` after_save :increment_user_posts_counter,`**

 ```
 after_save :increment_user_posts_counter

private

    def increment_user_posts_counter
      author.increment!(:posts_counter)
    end
```